lua/spellsitter.lua

```diff
@@ -149,8 +149,10 @@ local function enabled(bufnr, winid)
   if not vim.wo[winid].spell then
     return false
   end
-  if not vim.treesitter.highlighter.active[bufnr] then
-    return false
+  if not vim.treesitter == nil then
+    if not vim.treesitter.highlighter.active[bufnr] then
+      return false
+    end
```

Comment:
This should be:

```lua
if not vim.treesitter then
  return false
end
if not vim.treesitter.highlighter.active[bufnr] then
  return false
end
,,,

Reply:
I thought the same too, but if i try it i get the same error as above.

Reply:
Exactly as I've written it?

If vim.treesitter is nil then we should disable spellsitter for the buffer.

Reply:
Force pushed a variant that seems to work.

    if vim.treesitter ~= nil then
      return false
    end

Reply:
Sorry, that looks like a terrible idea, i didn't realise `~=` was not equals (thinking of regex...).

Reply:
That will disable spellsitter if `vim.treesitter` is defined which we don't want.

Reply:
I don't really know what's happening. `vim.treesitter` is showing as a table, but if the program tries to access `vim.treesitter.highligher` or `vim.treesitter["highlighter"]` it gets the "attempt to call a nil value" error.

Reply:
You might need to check `vim.treesitter.highlighter` is defined instead. I think `vim.treesitter` is always defined.

Reply:
Last resort would be to wrap the original check in `pcall`.